File: Generated_Codes_Check/Results/java/281.java

Testcase 1:
Input
3
3 1 2 2
3 4
2 2
3 3 3
3 4
2 2
3 3 1
40 50
20 30
Output
LuckyChef
UnluckyChef
UnluckyChef

Solution:

import java.util.*;
import java.lang.*;
import java.io.*;
class GFG
 {
	public static void main (String[] args)
	 {
	    Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
		while(t-- > 0)
		{
		    int x = sc.nextInt();
		    int y = sc.nextInt();
		    int k = sc.nextInt();
		    int n = sc.nextInt();
		    int p[] = new int[n];
		    int c[] = new int[n];
		    for(int i=0;i<n;i++)
		    {
		        p[i] = sc.nextInt();
		        c[i] = sc.nextInt();
		    }
		    int rem = x-y;
		    boolean flag = false;
		    for(int i=0;i<n;i++)
		    {
		        if(p[i]>=rem && c[i]<=k)
		        {
		            flag = true;
		            break;
		        }
		    }
		    if(flag)
		    {
		        System.out.println("LuckyChef");
		    }
		    else
		    {
		        System.out.println("UnluckyChef");
		    }
		}
	 }
}

File: Scripts/generated_codes/python/codechef_responses/281.py
'''
Example case 4. Chef cannot use the old notebook, because the old notebook doesn't contain sufficient number of pages.

Testcase 1 :
5 3 2 2
3 4
2 2
Output :
LuckyChef

Testcase 2 :
5 2 1 2
3 4
2 2
Output :
UnluckyChef

Testcase 3 :
3 3 2 2
2 3
3 3
Output :
UnluckyChef

Testcase 4 :
3 2 2 2
3 4
2 2
Output :
UnluckyChef

Testcase 5 :
1 1 1 2
1 2
2 3
Output :
LuckyChef

Testcase 6 :
1 1 1 1
1 3
Output :
UnluckyChef
'''
T=int(input())
for i in range(T):
    X,Y,K,N=list(map(int,input().split()))
    flag=0
    for j in range(N):
        P,C=map(int,input().split())
        if(X<=Y+P and C<=K):
            flag=1
            break
    if(flag==1):
        print("LuckyChef")
    else:
        print("UnluckyChef")